pkg/cmd/connector/update/update.go

```diff
@@ -35,85 +41,184 @@ func NewUpdateCommand(f *factory.Factory) *cobra.Command {
 		Short:   f.Localizer.MustLocalize("connector.update.cmd.shortDescription"),
 		Long:    f.Localizer.MustLocalize("connector.update.cmd.longDescription"),
 		Example: f.Localizer.MustLocalize("connector.update.cmd.example"),
-		Hidden:  true,
 		Args:    cobra.NoArgs,
+		Hidden:  true,
 		RunE: func(cmd *cobra.Command, args []string) error {
-
 			validOutputFormats := flagutil.ValidOutputFormats
 			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, validOutputFormats...) {
 				return flagutil.InvalidValueError("output", opts.outputFormat, validOutputFormats...)
 			}
 
-			return runUpdateCommand(opts)
+			return runUpdate(opts)
 		},
 	}
-
-	flags := connectorcmdutil.NewFlagSet(cmd, f)
-	flags.StringVar(&opts.file, "file", "", f.Localizer.MustLocalize("connector.file.flag.description"))
-	flags.AddConnectorID(&opts.id)
+	flags := flagutil.NewFlagSet(cmd, f.Localizer)
+	flags.StringVar(&opts.name, "name", "", f.Localizer.MustLocalize("connector.flag.name.description"))
+	flags.StringVar(&opts.namespaceID, "namespace-id", "", f.Localizer.MustLocalize("connector.flag.kafkaID.description"))
+	flags.StringVar(&opts.kafkaID, "kafka-id", "", f.Localizer.MustLocalize("connector.flag.namespaceID.description"))
 	flags.AddOutput(&opts.outputFormat)
 
 	return cmd
-
 }
 
-func runUpdateCommand(opts *options) error {
-	f := opts.f
+func runUpdate(opts *options) error {
+
+	conn, err1 := opts.f.Connection()
+	if err1 != nil {
+		return err1
+	}
+
+	api := conn.API()
+
+	connector, err2 := contextutil.GetCurrentConnectorInstance(&conn, opts.f)
+	if err2 != nil {
+		connector, _ = connectorutil.InteractiveSelect(conn, opts.f)
+	}
+
+	patch, err3 := setValuesInInteractiveMode(connector, opts, &conn)
+	if err3 != nil {
+		return err3
+	}
+
+	connectorSpecification, err4 := askForConnectorDetails(&api, connector, opts)
```

Comment:
If I want to update just one element like connector name would i need to fill the details?